test/NonSiloTests/Serialization/BuiltInSerializerTests.cs

```diff
@@ -727,9 +729,9 @@ public void Serialize_Uri_Multithreaded(SerializerToUse serializerToUse)
         [InlineData(SerializerToUse.NoFallback)]
         public void Serialize_GrainReference(SerializerToUse serializerToUse)
         {
-            InitializeSerializer(serializerToUse);
+            var environment = InitializeSerializer(serializerToUse);
             GrainId grainId = GrainId.NewId();
-            GrainReference input = GrainReference.FromGrainId(grainId);
+            GrainReference input = environment.InternalGrainFactory.GetGrain(grainId);
 
             object deserialized = OrleansSerializationLoop(input);
```

Comment:
I'll add the check. There are other tests which indirectly test this by making a call on the deserialized grain